pkg/resource/replication_group/custom_update_api_test.go

```diff
@@ -392,6 +393,56 @@ func TestCustomModifyReplicationGroup_NodeGroup_available(t *testing.T) {
 	})
 }
 
+func TestCustomModifyReplicationGroup_Scaling_Async_Rollback(t *testing.T) {
+	assert := assert.New(t)
+	t.Run("ScaleDownRollback=Diff", func(t *testing.T) {
+		desired := provideResource()
+		latest := provideResource()
+		rgId := "RGID"
+		desired.ko.Spec.ReplicationGroupID = &rgId
+		latest.ko.Spec.ReplicationGroupID = &rgId
+		desired.ko.ObjectMeta.Annotations = make(map[string]string)
+		desiredCacheNodeType := "cache.t3.micro"
+		currentCacheNodeType := "cache.m5.large"
+		desired.ko.Annotations[AnnotationLastRequestedCNT] = desiredCacheNodeType
+		desired.ko.Spec.CacheNodeType = &desiredCacheNodeType
+
+		rm := provideResourceManager()
+
+		var delta ackcompare.Delta
+		delta.Add("Spec.CacheNodeType", currentCacheNodeType, desiredCacheNodeType)
+
+		var ctx context.Context
+		res, err := rm.CustomModifyReplicationGroup(ctx, desired, latest, &delta)
+		assert.Nil(res)
+		assert.NotNil(err)
+		assert.Equal("InvalidParameterCombination: Cannot update CacheNodeType, Please refer to Events for more details", err.Error())
+	})
+
+	t.Run("ScaleInRollback=Diff", func(t *testing.T) {
+		desired := provideResource()
+		latest := provideResource()
+		rgId := "RGID"
+		desired.ko.Spec.ReplicationGroupID = &rgId
+		latest.ko.Spec.ReplicationGroupID = &rgId
+		desired.ko.ObjectMeta.Annotations = make(map[string]string)
+
+		desiredNodeGroup := int64(4)
+		currentNodeGroup := int64(3)
+		desired.ko.Annotations[AnnotationLastRequestedNNG] = strconv.Itoa(int(desiredNodeGroup))
+		desired.ko.Spec.NumNodeGroups = &desiredNodeGroup
+		rm := provideResourceManager()
+
+		var delta ackcompare.Delta
+		delta.Add("Spec.NumNodeGroups", currentNodeGroup, desiredNodeGroup)
+
+		var ctx context.Context
```

Comment:
nit: i know that `CustomModifyReplicationGroup` doesn't touch the `ctx` variable here, but ideally we should use `context.TODO()`